src/fundamental/macro-fundamental.h

```diff
@@ -41,7 +41,12 @@
 #define CONCATENATE(x, y) XCONCATENATE(x, y)
 
 #ifdef SD_BOOT
-#define assert(expr) do {} while (false)
+        #ifdef EFI_DEBUG
```

Comment:
why introduce a new macro for this? Sounds like we should just use NDEBUG like regular libc. It has an inverted sense, , which i think is good, since it means by default asserts are n, as they should be

Reply:
Mostly because that's what the (unused) gnu-efi assert code uses.

Reply:
good point

Reply:
Shall I leave this then. Use both?

Reply:
given they are inverted i don't see how we could use both...

Reply:
i guess that means as long as we use gnuefi we should probably stick to its semantics on it. but please add comment about the fact that ndebug is not used

Reply:
It seems meson default for b_ndebug is false, so even release builds get asserts. (Arch doesn't change this either it seems). While I only set EFI_DEBUG on debug builds. I'm rather inclined to make asserts depends on NDEBUG to match behaviors. Somehow I feel somewhere a ball was dropped regarding ndebug...